src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -448,21 +448,27 @@ public void Dispose()
             }
         }
 
-        internal class NodeModificationUndoHelper : IDisposable
+        internal class ModelModificationUndoHelper : IDisposable
         {
-            private readonly NodeModel node;
+            private readonly List<ModelBase> models;
             private readonly UndoRedoRecorder recorder;
             private readonly Dictionary<Guid, XmlElement> existingConnectors;
             private readonly Dictionary<Guid, ConnectorModel> remainingConnectors;
 
-            public NodeModificationUndoHelper(UndoRedoRecorder recorder, NodeModel node)
+            public ModelModificationUndoHelper(UndoRedoRecorder recorder, ModelBase model)
+                : this(recorder, new [] { model })
```

Comment:
Just curious why has `NodeModel` been replaced with `ModelBase`. Is it to support `NoteModel` updates? If so, why was it not handled earlier?